include/mbedtls/check_config.h

```diff
@@ -683,6 +683,17 @@
 #error "MBEDTLS_HAVE_INT32/MBEDTLS_HAVE_INT64 and MBEDTLS_HAVE_ASM cannot be defined simultaneously"
 #endif /* (MBEDTLS_HAVE_INT32 || MBEDTLS_HAVE_INT64) && MBEDTLS_HAVE_ASM */
 
+#if defined(MBEDTLS_SSL_PROTO_SSL3) && defined(MBEDTLS_DEPRECATED_WARNING)
+#warning "MBEDTLS_SSL_PROTO_SSL3 is deprecated and will likely be removed in a future version of Mbed TLS"
+#endif /* MBEDTLS_SSL_PROTO_SSL3 && MBEDTLS_DEPRECATED_WARNING */
+
+#if defined(MBEDTLS_SSL_SRV_SUPPORT_SSLV2_CLIENT_HELLO) && \
+    defined(MBEDTLS_DEPRECATED_WARNING)
+#warning "MBEDTLS_SSL_SRV_SUPPORT_SSLV2_CLIENT_HELLO is deprecated and will likely be removed in a future version of Mbed TLS"
+#endif /* MBEDTLS_SSL_SRV_SUPPORT_SSLV2_CLIENT_HELLO &&
+        * MBEDTLS_DEPRECATED_WARNING
+        */
```

Comment:
I am not sure about this. The problem is that we are deprecating here a feature rather than a function, which is what MBEDTLS_DEPRECATED_WARNING was meant for. So if I enable SSLv2 or SSLv3 and MBEDTLS_DEPRECATED_WARNING I will not get any warnings from use of deprecated functions.

To make it more explicit, I decided to add these two check to the check_config.h. Of course, the problem is that #warning does not work for all toolchains (but MBEDTLS_DEPRECATED_WARNING does say it only works on clang and gcc). Also, this changes the MBEDTLS_DEPRECATED_WARNING a little bit...

Reply:
I like this, and I think that's the right way to do it. As you say, as a user if I enable `MBEDTLS_DEPRECATED_WARNING`, I don't want to be able to use anything deprecated without getting a warning. We should probably just adjust the description of this option to say "function or feature" instead of just "function".